extras/indexing/src/main/java/org/apache/rya/indexing/entity/storage/mongo/MongoEntityStorage.java

```diff
@@ -242,4 +281,46 @@ private static Bson makeExplicitTypeFilter(final RyaURI typeId) {
 
         return Stream.of(dataTypeFilter, valueFilter);
     }
+
+    private boolean detectDuplicates(final Entity entity) throws EntityStorageException {
+        boolean hasDuplicate = false;
+        if (duplicateDataDetector.isDetectionEnabled()) {
+            if (mongoTypeStorage == null) {
+                mongoTypeStorage = new MongoTypeStorage(mongo, ryaInstanceName);
+            }
+            final Builder builder = new Builder();
+            builder.setSubject(entity.getSubject());
+            boolean abort = false;
+            for (final RyaURI typeRyaUri : entity.getExplicitTypeIds()) {
+                Optional<Type> type;
+                try {
+                    type = mongoTypeStorage.get(typeRyaUri);
+                } catch (final TypeStorageException e) {
+                    throw new EntityStorageException("Unable to get entity type: " + typeRyaUri, e);
+                }
+                if (type.isPresent()) {
+                    final ConvertingCursor<TypedEntity> cursor = search(Optional.empty(), type.get(), Collections.emptySet());
+                    while (cursor.hasNext()) {
+                        final TypedEntity typedEntity = cursor.next();
```

Comment:
If I recall correctly, a TypedEntity is one part of an Entity and an Entity is comprised of possibly several TypedEntities.  (i.e an Entity could be made up of a person TypedEntity and an employee TypedEntity).  So, TypedEntity shouldn't extend Entity.